xocto/storage/s3_select.py

```diff
@@ -58,12 +58,9 @@ def to_dict(self) -> dict[str, Any]:
         temp_dict = {k: v for k, v in dataclasses.asdict(self).items() if v}
 
         for k, v in temp_dict.items():
-            if (
-                isinstance(v, FileHeaderInfo)
-                or isinstance(v, QuoteFields)
-                or isinstance(v, CompressionType)
-            ):
+            if type(v) in [FileHeaderInfo, QuoteFields, CompressionType]:
```

Comment:
xocto currently supports Python 3.9, which doesn’t include switch/case - those comes in Python 3.10. So this was the right move. For now. :wink: